### PR TITLE
espresso: init at 2.4

### PR DESCRIPTION
**Pull request by pineapplehunter**
## Description of changes

- Added espresso as a new package

The license is set ro unfree, since the source on github does not provide a licence. See: https://github.com/chipsalliance/espresso/issues/4

cc: @sequencer 

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/es/espresso/package.nix [hunk 1]
```diff
@@ -0,0 +1,43 @@
+{ lib, fetchFromGitHub, cmake, stdenv, ninja, nix-update-script }:
+stdenv.mkDerivation rec {
+  pname = "espresso";
+  version = "2.4";
+  src = fetchFromGitHub {
+    owner = "chipsalliance";
+    repo = pname;
+    rev = "v${version}";
+    sha256 = "sha256-z5By57VbmIt4sgRgvECnLbZklnDDWUA6fyvWVyXUzsI=";
```
Comment (NickCao) [inline pkgs/by-name/es/espresso/package.nix:9]:
```suggestion
    hash = "sha256-z5By57VbmIt4sgRgvECnLbZklnDDWUA6fyvWVyXUzsI=";
```

File: pkgs/by-name/es/espresso/package.nix [hunk 2]
```diff
@@ -0,0 +1,43 @@
+{ lib, fetchFromGitHub, cmake, stdenv, ninja, nix-update-script }:
+stdenv.mkDerivation rec {
+  pname = "espresso";
+  version = "2.4";
+  src = fetchFromGitHub {
+    owner = "chipsalliance";
+    repo = pname;
```
Comment (NickCao) [inline pkgs/by-name/es/espresso/package.nix:7]:
```suggestion
    repo = "espresso";
```

File: pkgs/by-name/es/espresso/package.nix [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+    rev = "v${version}";
+    sha256 = "sha256-z5By57VbmIt4sgRgvECnLbZklnDDWUA6fyvWVyXUzsI=";
+  };
+
+  nativeBuildInputs = [ cmake ninja ];
```
Comment (NickCao) [inline pkgs/by-name/es/espresso/package.nix:12]:
```suggestion
  nativeBuildInputs = [ cmake ];
```

`ninja` is only faster on incremental builds.

File: pkgs/by-name/es/espresso/package.nix [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+
+  enableParallelBuilding = true;
```
Comment (NickCao) [inline pkgs/by-name/es/espresso/package.nix:14]:
```suggestion
```
`cmake` already sets this by default.

File: pkgs/by-name/es/espresso/package.nix [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+  doCheck = true;
+
+  outputs = [ "out" "man" ];
+
+  passthru.updateScript = nix-update-script { };
+
+  meta = with lib;{
+    description = "Espresso is a Multi-valued PLA minimization.";
```
Comment (NickCao) [inline pkgs/by-name/es/espresso/package.nix:23]:
```suggestion
    description = "Multi-valued PLA minimization";
```

File: pkgs/by-name/es/espresso/package.nix [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+      two-valued (or multiple-valued) Boolean function, and produces a
+      minimal equivalent representation.  The algorithms used are new and
+      represent an advance in both speed and optimality of solution in
+      heuristic Boolean minimization.
+    '';
+    homepage = "https://github.com/chipsalliance/espresso";
+    maintainers = with maintainers;[ pineapplehunter ];
+    mainProgram = "espresso";
+    platforms = lib.platforms.all;
+    sourceProvenance = with lib.sourceTypes;[ fromSource ];
```
Comment (NickCao) [inline pkgs/by-name/es/espresso/package.nix:36]: No need to mark sourceProvenance explicitly for source builds.
```suggestion
```

**Comment by pineapplehunter** [comment]
@NickCao Thanks for the review!
Updated the commit.

**Comment by NickCao** [comment]
Result of `nixpkgs-review pr 272597` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>espresso</li>
    <li>espresso.man</li>
  </ul>
</details>

**Pull request merged**